### PR TITLE
Implemented side bar, added dynamic CSS styling on mouse hover, cleaned up code.

### PR DESCRIPTION
**Pull request by John-Michael-Long**

### CODE REVIEW

**Comment by John-Michael-Long** [comment]
@zmattingly can you review my PR.

File: client/components/GameDetailsSideBar.jsx [hunk 1]
```diff
@@ -2,7 +2,20 @@ import React from 'react';
...
 const GameDetailsSideBar = props => (
   <div className="side-bar">
-    Hello from side bar
+    <div className="side-bar-game-info-text">
+      <div className="side-bar-title">{props.game.title}</div>
+      <div>{props.game.description}</div>
+      <div>{props.game.categories}</div>
+    </div>
+    <div className="screenshots">
+      {props.game.screenshots ? props.game.screenshots.map((fileName, index) =>
```
Comment (zmattingly) [inline client/components/GameDetailsSideBar.jsx]: For conditional rendering I think it's generally preferred to assign the result of your `.map()` to a variable above the return portion of your stateless Component function, then reference that variable inside the returned markup to make it clear exactly what should be rendered.

This was a bit confusing to me to read at first because I had to lookup whether `.map()` would return an array of your `` elements or if it was modifying the array in-place. I think separating this out will just make this a bit more readable.

File: client/components/GameList.jsx [hunk 2]
```diff
@@ -3,13 +3,13 @@ import GameListEntry from './GameListEntry.jsx';
...
-    <div className="game-list-heading">List of Related Games!</div>
     {props.games.map((game, index) => (
       <GameListEntry 
         game={game} 
         key={index}
         handleGameMouseEntry={props.handleGameMouseEntry}
         handleGameMouseExit={props.handleGameMouseExit}
+        selectedGame={props.selectedGame}
       />
     ))}
```
Comment (zmattingly) [inline client/components/GameList.jsx:14]: Might want to do the same thing here as I recommended in the `GameDetailsSideBar.jsx` file where you assign the map results to a variable, but I think in this case this is concise enough and doesn't include a conditional so you're probably fine if you want to keep it this way.

File: client/index.jsx [hunk 3]
```diff
@@ -22,9 +23,9 @@ class App extends React.Component {
       type: 'GET',
       url: 'http://127.0.0.1:3002/games',
```
Comment (zmattingly) [inline client/index.jsx]: You should simply use `/games` as the url option here. Referencing `localhost` or `127.0.0.1` will not work when you go to deploy this later to AWS or Heroku. As long as the client is connecting to the main Express server on Port 3002, that same hostname and port will be used for your API calls.

File: client/index.jsx [hunk 4]
```diff
@@ -33,8 +34,10 @@ class App extends React.Component {
     });
   }
 
-  handleGameMouseEntry() {
-    console.log('entered mouse entry handler');
+  handleGameMouseEntry(game) {
+    this.setState({
+      selectedGame: game,
```
Comment (zmattingly) [inline client/index.jsx:39]: No comma needed here, linter will probably complain about it.

File: client/components/GameListEntry.jsx [hunk 5]
```diff
@@ -2,8 +2,8 @@ import React from 'react';
 
 const GameListEntry = props => (
   <div 
-    className="game-entry" 
-    onMouseEnter={props.handleGameMouseEntry}
+    className={props.game.title === props.selectedGame.title ? 'current-game' : 'game-entry'}
+    onMouseEnter={(game => props.handleGameMouseEntry(game)).bind(null, props.game)}
```
Comment (zmattingly) [inline client/components/GameListEntry.jsx]: It's kinda hard to parse what's going on here, I had to do a lot of back and forth through files to sort out what was happening.

Rather than using another `.bind()` around an anonymous function enclosed in parens, I'd recommend just creating a named wrapper arrow function above your `const GameListEntry` assignment which simply accepts a `game` argument and a `callback`. Then just have it return a function where you execute the callback (which you'd be passing in as `props.handleGameMouseEntry`) with `game` as the sole argument.

If this pattern is used elsewhere you could just stick it in a module and import it where needed across Components.

File: public/style.css [hunk 6]
```diff
@@ -1,57 +1,81 @@
+body {
```
Comment (zmattingly) [inline public/style.css:1]: What's up with two `style.css` files? I think you probably want to add another public Express directory using `express.static()` and point it to `client/dist/`

**Review by zmattingly — CHANGES_REQUESTED** [review]
Heyo, got a few comments here for review.

In addition to these, I also went to clone down your repo and run it locally but there are no instructions on how to get this up and running. Not critical for this PR but can you fill out the setup steps in your README.md for the next one?

File: client/components/GameDetailsSideBar.jsx (hunk 1 above)
Comment (zmattingly) [reply client/components/GameDetailsSideBar.jsx]: Generally in the docs here they are using this approach where they assign the map results to a variable: https://reactjs.org/docs/lists-and-keys.html

There is section at the bottom which outlines this case where you inline the map() result, but it notes that it's up to you to decide which is more readable.

File: client/index.jsx (hunk 3 above)
Comment (John-Michael-Long) [reply client/index.jsx]: implemented change

File: client/index.jsx (hunk 4 above)
Comment (John-Michael-Long) [reply client/index.jsx:39]: I thought so too, however linter was actually requesting the comma

File: client/components/GameListEntry.jsx (hunk 5 above)
Comment (John-Michael-Long) [reply client/components/GameListEntry.jsx]: implemented change

File: client/index.jsx (hunk 3 above)
Comment (John-Michael-Long) [reply client/index.jsx]: implemented change

File: client/components/GameDetailsSideBar.jsx (hunk 1 above)
Comment (John-Michael-Long) [reply client/components/GameDetailsSideBar.jsx]: implemented change

File: client/components/GameList.jsx (hunk 2 above)
Comment (John-Michael-Long) [reply client/components/GameList.jsx:14]: deferred change

File: public/style.css (hunk 6 above)
Comment (John-Michael-Long) [reply public/style.css:1]: this is a result of the eslint webpack plugin,  I will remove these files from the repo and add to gitignore

**Comment by John-Michael-Long** [comment]
@zmattingly implemented changes from PR review feedback. Please merge PR if acceptable.

File: client/index.jsx (hunk 4 above)
Comment (zmattingly) [reply client/index.jsx:39]: Oh weird. ¯\\\_(ツ)\_/¯

**Review by zmattingly — APPROVED** [review]

**Comment by zmattingly** [comment]
Looks good.
